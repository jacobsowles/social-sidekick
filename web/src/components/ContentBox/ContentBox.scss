.content-box {
  background: #ffffff;
  border: 1px solid rgba(0, 0, 0, 0.1);
  margin-bottom: 20px;
  padding: 20px;

  p {
    margin-bottom: 20px;

    &:last-child {
      margin-bottom: 0;
    }
  }

  .loading-spinner {
    width: 100%;
  }
}
