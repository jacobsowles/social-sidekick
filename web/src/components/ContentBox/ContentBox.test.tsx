import React from 'react';
import { init, shallow, shallowTopLevelElement, toJson } from '@tests/component-test-base';
import ContentBox, { ContentBoxProps } from './ContentBox';

describe('ContentBox', () => {
  let props: ContentBoxProps;

  init(
    () => (
      <ContentBox {...props}>
        <p>Test</p>
      </ContentBox>
    ),
    'div'
  );

  beforeEach(() => {
    props = {
      className: undefined
    };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });

  describe('the rendered div element', () => {
    it('should contain all rendered child components', () => {
      shallowTopLevelElement()
        .children()
        .should.deep.equal(shallow().children());
    });
  });

  describe('when `className` is defined', () => {
    beforeEach(() => {
      props.className = 'another-class';
    });

    it('should include the specified class name', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.contain('another-class');
    });

    it('should include the content-box class name', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.contain('content-box');
    });
  });

  describe('when `className` is undefined', () => {
    it('should include the content-box class name', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.contain('content-box');
    });
  });
});
