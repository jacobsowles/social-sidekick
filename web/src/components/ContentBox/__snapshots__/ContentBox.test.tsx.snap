// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBox should render the component without crashing 1`] = `
<div
  className="content-box"
>
  <p>
    Test
  </p>
</div>
`;
